src/main/java/org/mdbenefits/app/preparers/IncomeDetailsPreparer.java

```diff
@@ -39,28 +39,45 @@ public Map<String, SubmissionField> prepareSubmissionFields(Submission submissio
 
     private Map<String, SubmissionField> prepareIncome(Submission submission) {
         Map<String, SubmissionField> fields = new HashMap<>();
-        var income = (List<Map<String, Object>>) submission.getInputData().get("income");
-        if (income != null) {
-            int iterator = 1;
-            for (int i = 0; i < income.size(); i++) {
-                Map<String, Object> incomeDetails = income.get(i);
+        Map<String, Object> inputData = submission.getInputData();
+        var income = (List<Map<String, Object>>) inputData.get("income");
+
+        boolean hasIncome = income != null && !income.isEmpty();
+        fields.put(
+                "householdHasEarnedIncome",
+                new SingleField("householdHasEarnedIncome", hasIncome ? "true" : "false", null));
 
+        if (income != null) {
+            int i = 1;
+            for (Map<String, Object> incomeDetails : income) {
                 var employeeName = incomeDetails.get("householdMemberJobAdd");
+                if (employeeName.equals("you")) {
```

Comment:
I can't recall if we stored the applicant name as "you" before or if we translated it in an action after the page.  @analoo do you know? Given that you have to add this logic, perhaps we already had changed "you" -> "firstname, lastname" before we got this far.  There are some pieces of code that make me think we leave it as "you", so perhaps this is just fine.

Reply:
It was filling the PDF in as "you" before, which was why I added this if statement.

Reply:
From what I recall we use 'you' to refer to the candidate so this update makes sense to me.